src/table/components/head/HeadCellMenu.tsx

```diff
@@ -46,15 +65,89 @@ export default function HeadCellMenu({ column, tabIndex }: HeadCellMenuProps) {
                   embedListbox();
                 },
                 icon: <Search />,
-                isDisabled: false,
+                enabled: true,
+              },
+            ],
+            [
+              {
+                id: 1,
+                itemTitle: translator.get('SNTable.MenuItem.SelectAll'),
+                onClick: async () => {
+                  await fieldInstance?.selectAll();
+                  setOpenMenuDropdown(false);
```

Comment:
any reason to await this? We might as well close the menu directly, no need for the selection call to finish?

Reply:
awaiting because all of those function returns a promise in the type file. also of course it is async call. I'll move state change above it 👍